### PR TITLE
Remove SS_NO_TICK_CHECK

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
This is a whopper, will need @MrStonedOne to review.
In theory this flag shouldn't be needed and my own testing didn't find any issues, but I'd feel better with a lengthy test merge.

## Why It's Good For The Game
Might make nightshift subsystem lag less, MC loop has less stuff to check.

### CODE REVIEW

**Review by MrsTonedOne — APPROVED** [review]
Do make sure it compiles with the flag un-defined to ensure you got every reference.

File: code/controllers/subsystem/nightshift.dm [hunk 1]
```diff
@@ -41,15 +44,19 @@ SUBSYSTEM_DEF(nightshift)
...
+/datum/controller/subsystem/nightshift/proc/update_nightshift(active, announce = TRUE, resumed = FALSE)
+	if(!resumed)
+		currentrun = GLOB.apcs_list.Copy()
+		nightshift_active = active
+		if(announce)
+			if (active)
+				announce("Good evening, crew. To reduce power consumption and stimulate the circadian rhythms of some species, all of the lights aboard the station have been dimmed for the night.")
+			else
+				announce("Good morning, crew. As it is now day time, all of the lights aboard the station have been restored to their former brightness.")
+	for(var/A in currentrun)
```
Comment (Mothblocks) [inline code/controllers/subsystem/nightshift.dm:56]: Make this an `as anything` loop--the type is redefined inside.

File: code/__DEFINES/MC.dm [hunk 2]
```diff
@@ -37,6 +37,7 @@
 #define SS_BACKGROUND 4
 
 /// subsystem does not tick check, and should not run unless there is enough time (or its running behind (unless background))
+/// DEPRECIATED
 #define SS_NO_TICK_CHECK 8
```
Comment (Mothblocks) [inline code/__DEFINES/MC.dm:41]: Agree with MSO, why is this still defined?

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/__DEFINES/MC.dm (hunk 2 above)
Comment (spookydonut) [reply code/__DEFINES/MC.dm:41]: will remove

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**